docs/functions.md

```diff
@@ -23,8 +23,12 @@
   - `replace` _[boolean]_ - create or replace function
   - `window` _[boolean]_ - window function
   - `behavior` _[string]_ - `IMMUTABLE`, `STABLE`, or `VOLATILE`
-  - `onNull` _[boolean]_ - `RETURNS NULL ON NULL INPUT`
+  - `strict` _[boolean]_ - `RETURNS NULL/CALLED ON NULL INPUT`
+  - `onNull` _[string]_ (deprecated) - alias for `strict`, also accepts `NULL` or `CALLED`
```

Comment:
I do not think we need another parameter and value doing the same.
`CALLED ON NULL INPUT` is the default, you can switch to `STRICT`/`RETURNS NULL ON NULL INPUT` if you want.
If there will be more options in the future, we will revisit it

Reply:
The problem I had with `onNull` was that a boolean `onNull` was not really descriptive. To me, `onNull: true` looks like it means "called on null" whereas `onNull: false` would mean "not called on null", right? - except that it's exactly the opposite.
The new way of writing `onNull: 'NULL'` or `strict: true` is much less confusing, at least to me.

Reply:
Well for me it is now quite confusing as you extend the type with string and deprecate it and introduced a new field that behaves the same. I have to think about...

Reply:
I think I would prefer having *either* `onNull: 'CALLED' | 'NULL'` *or* `strict: boolean`, (in addition to supporting the legacy boolean `onNull`). Not sure why I added both in this PR, probably just to demonstrate my ideas.